cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/DBSource.java

```diff
@@ -0,0 +1,94 @@
+import co.cask.cdap.templates.etl.common.DBRecord;
+import co.cask.cdap.templates.etl.common.Properties;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.db.DBConfiguration;
+import org.apache.hadoop.mapreduce.lib.db.DBInputFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * Batch source to read from a Database table
+ */
+public class DBSource extends BatchSource<LongWritable, DBRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(DBSource.class);
+
+  @Override
+  public void configure(StageConfigurer configurer) {
+    configurer.setDescription("Batch source for database");
+    configurer.addProperty(new Property(Properties.DB.DRIVER_CLASS, "Driver class to connect to the database", true));
+    configurer.addProperty(
+      new Property(Properties.DB.CONNECTION_STRING, "JDBC connection string including database name", true));
+    configurer.addProperty(new Property(Properties.DB.TABLE_NAME, "Table name to import", true));
+    configurer.addProperty(
+      new Property(Properties.DB.COLUMNS,
+                   "Comma-separated list of columns to import. When unspecified, defaults to all columns.", false));
+  }
+
+  @Override
+  public void configurePipeline(ETLStage stageConfig, PipelineConfigurer pipelineConfigurer) {
+    Map<String, String> properties = stageConfig.getProperties();
+    String dbDriverClass = properties.get(Properties.DB.DRIVER_CLASS);
+    String dbConnectionString = properties.get(Properties.DB.CONNECTION_STRING);
+    String dbTableName = properties.get(Properties.DB.TABLE_NAME);
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(dbDriverClass), "dbDriverClass cannot be null");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(dbConnectionString), "dbConnectionString cannot be null");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(dbTableName), "dbTableName cannot be null");
+  }
+
+  @Override
+  public void prepareJob(BatchSourceContext context) {
+    Map<String, String> runtimeArguments = context.getRuntimeArguments();
+    String dbDriverClass = runtimeArguments.get(Properties.DB.DRIVER_CLASS);
+    String dbConnectionString = runtimeArguments.get(Properties.DB.CONNECTION_STRING);
+    String dbTableName = runtimeArguments.get(Properties.DB.TABLE_NAME);
+    String dbColumns = runtimeArguments.get(Properties.DB.COLUMNS);
+
+    LOG.debug("dbTableName = {}; dbDriverClass = {}; dbConnectionString = {}; dbColumns = {}",
+              dbTableName, dbDriverClass, dbConnectionString, dbColumns);
+
+    String countQueryColumn;
+    if (dbColumns == null) {
+      dbColumns = "*";
+      countQueryColumn = "*";
+    } else {
+      countQueryColumn = dbColumns.substring(0, dbColumns.indexOf(","));
```

Comment:
Seems like this logic should be in `configurePipeline` so that it can fail the creation if the config is not right (e.g. dbColumns does not have ",").